src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -2652,23 +2654,31 @@ The Signal API is available inside ProActive Tasks through the *signalapi* <<_gl
 
 It is easy to use, and offers a set of basic operations on signals summarized as follows:
 
- * *readyForSignal*: declares that the current job is ready to receive and handle one or more signals.
+ * *readyForSignal*: declares that the current job is ready to receive and handle one or more signals with or without variables.
    This operation constructs new signals (by adding the prefix `ready_` to the input signals), then adds them to job signals.
    For example, if a job declares ready for the signal `soft_stop`, the signal `ready_soft_stop` is added to the set of job signals.
    Additionally, if the input signal (e.g., `soft_stop`) already exists among the job signals (which means that it has been sent and handled previously),
     the operation ready for signal automatically removes it, thus the job is ready again to receive and handle the same signal.
 
- * *sendSignal*, *sendManySignals*: adds one or more signals given as input to the current job signals.
+ * *sendSignal*, *sendManySignals*: adds one or more signals with or without variables given as input to the current job signals.
    This operation removes the corresponding ready signals if they already exist. For instance, sending `soft_stop` removes the signal
    `ready_soft_stop` if it already exists among the job signals.
 
- * *checkForSignals*: verifies whether one or more signals exist among the set of job signals.
+ * *checkForSignals*: verifies whether all of the signals given as input exist among the set of job signals.
 
  * *waitFor*, *waitForAny*: blocking instruction that waits until one signal (among those given as input) is added to the set of job signals.
 
  * *removeSignal*, *removeManySignals*: removes one or more signals from the set of job signals.
 
- * *getJobSignals*: returns the current job signals.
+ * *getJobSignals*: returns the current job signals and variables.
+
+The specified operations can be performed for both *signals* and *signals with variables*.
+
+A signal can have input variables and updated variables.
+
+The variables of a signal have predefined values called input variables that can be set using readyForSignal operation.
+
+The updated variables of a signal can be set using sendSignal and sendManySignals operations.
```

Comment:
I think these 4 sentences are not understandable, they include implementation details (which should not be present in the documentation).

I would remove them altogether.